core/chains/evm/client/client.go

```diff
@@ -197,6 +203,108 @@ func (client *client) HeaderByHash(ctx context.Context, h common.Hash) (*types.H
+		// success (even though the transaction will never confirm) and hand
+		// off to the ethConfirmer to bump gas periodically until we _can_ get
+		// it in
+		//
+		// Confirmer: Most likely scenario here is a parity node that is rejecting
+		// low-priced transactions due to mempool pressure.
+		// In that case, the safest thing to do is to pretend the transaction
+		// was accepted and continue the normal gas bumping cycle until we can
+		// get it into the mempool
+		client.logger.Infow("Transaction temporarily underpriced", "err", sendError.Error())
+		return clienttypes.Successful, err
+	}
+	if sendError.IsTerminallyUnderpriced() {
+		return clienttypes.Underpriced, err
+	}
+	if sendError.L2FeeTooLow() || sendError.IsL2FeeTooHigh() || sendError.IsL2Full() {
+		if client.pool.ChainType().IsL2() {
+			return clienttypes.FeeOutOfValidRange, err
+		}
+		return clienttypes.Unsupported, errors.Wrap(sendError, "this error type only handled for L2s")
+	}
+	if sendError.IsNonceTooHighError() {
+		// This error occurs when the tx nonce is greater than current_nonce + tx_count_in_mempool,
+		// instead of keeping the tx in mempool. This can happen if previous transactions haven't
+		// reached the client yet. The correct thing to do is assume success for now and let the
+		// eth_confirmer retry until the nonce gap gets filled by the previous transactions.
+		client.logger.Warnw("Transaction has a nonce gap.", "err", err)
+		return clienttypes.Retryable, err
+	}
+	if sendError.IsInsufficientEth() {
+		client.logger.Criticalw(fmt.Sprintf("Tx %x with type 0x%d was rejected due to insufficient eth: %s\n"+
+			"ACTION REQUIRED: Chainlink wallet with address 0x%x is OUT OF FUNDS",
+			tx.Hash(), tx.Type(), sendError.Error(), fromAddress,
+		), "err", sendError)
+		// NOTE: This bails out of the entire cycle and essentially "blocks" on
+		// any transaction that gets insufficient_eth. This is OK if a
+		// transaction with a large VALUE blocks because this always comes last
+		// in the processing list.
+		// If it blocks because of a transaction that is expensive due to large
+		// gas limit, we could have smaller transactions "above" it that could
+		// theoretically be sent, but will instead be blocked.
+		return clienttypes.InsufficientFunds, err
+	}
+	if sendError.IsTimeout() {
+		// In the case of timeout, we fall back to the backoff retry loop and
+		// subsequent tries ought to resend the exact same in-progress transaction
+		// attempt and get a definitive answer on what happened
+		return clienttypes.Retryable, errors.Wrapf(sendError, "timeout while sending transaction %s", tx.Hash().Hex())
+	}
+	if sendError.IsTxFeeExceedsCap() {
+		client.logger.Criticalw(fmt.Sprintf("Sending transaction failed: %s", label.RPCTxFeeCapConfiguredIncorrectlyWarning),
+			"etx", tx,
+			"err", sendError,
+			"id", "RPCTxFeeCapExceeded",
+		)
+		// Broadcaster: Note that we may have broadcast to multiple nodes and had it
```

Comment:
This comment too shouldn't be here. Instead move to corresponding places in the callers.

Reply:
We should mention in the comment in Broadcaster/Confirmer, what they do for each response they receive, and why. You're not changing the overall behavior in this PR, so I think broadcaster/confirmer are still doing same things. It might be that comments just need to provide more context.